### ISSUE TITLE
Test on Windows

### ISSUE THREAD

**Issue by StanleySweet** [brollb/0ad-rl-interface-js-sample#1]
Using
```js
async function runTests(address) {
    await resetScenario(address);
    i = 0
    while (i < 500) {
        await stepScenario(address);
        console.log('step!');
        ++i;
    }
}
```
And the following powershell code

```ps1
function get_elapsed_time($start_time)
{
    $now = Get-Date
    Write-Output "Time taken: $(($now).Subtract($start_time).Minutes) minutes(s) $(($now).Subtract($start_time).Seconds) second(s)"
}
$start_time = Get-Date
node run-scenario.js
get_elapsed_time($start_time)
```

500 steps take **56** seconds.

**Comment by brollb** [comment]
Wow. It takes me just under 8 secs on linux. Are you running 0 AD in nonvisual mode or with a GUI? (Not sure if it matters but want to make sure I am running it the same way!)

**Comment by StanleySweet** [comment]
I am using the command in the readme. Oh, my bad, I was rendering video in the background and it took a hit. Now that it finishes I'm back to 12s

**Comment by brollb** [comment]
Cool. Can you test it with the python equivalent (using `zero_ad`)? I just uploaded it as `run_scenario.py`! (I am getting just over 8 seconds.)

**Comment by StanleySweet** [comment]
Nope still slow. I think it's whatever is making requests in the python zero ad module

**Comment by brollb** [comment]
Thanks for checking! Out of curiousity, do you happen to have the actual time it takes for 500 requests?

It is pretty weird that python is so slow as it is using the standard urllib module in python (to avoid dependencies). I will dig into this some more and see what I can find out...

**Comment by StanleySweet** [comment]
I remember that there is a problem with flushing prints in the default lib as you can see here https://code.wildfiregames.com/D3213#142126 so maybe there is something dumb with the url stuff too

50 is Time taken: 1 minute(s) 45 second(s) 568 millisecond(s) so I assume 500 will be something like 10mins

**Comment by StanleySweet** [comment]
Any news?

**Comment by brollb** [comment]
Would you mind generating a flame chart of the execution on windows? It should be something like this:
```
python -m cProfile -o run_scenario.prof run_scenario.py
```
Then install and use `flameprof`:
```
pip install flameprof
python -m flameprof run_scenario.prof > run_scenario_flamechart.svg
```

I will try to replace the current approach with the `requests` library and see if that helps but it would be nice to know where specifically the slowdown is happening. (Thanks in advance!)

**Comment by brollb** [comment]
Also, you may want to change the [500 requests](https://github.com/brollb/0ad-rl-interface-js-sample/blob/master/run_scenario.py#L11) down to something like 2 or so. No need to make the script run much longer than necessary!

**Comment by StanleySweet** [comment]
[0ad-rl-interface-js-sample.zip](https://github.com/brollb/0ad-rl-interface-js-sample/files/5989299/0ad-rl-interface-js-sample.zip)

10 requests 23 seconds

**Comment by brollb** [comment]
Can you give `run_scenario_requests.py` a whirl? It uses a different python library (`requests`) so you will need to install it using `pip`. I prefer using the builtin version but maybe this has some way to get around the perf issue in windows...

**Comment by StanleySweet** [comment]
Fixed by replacing localhost:6000 by 127.0.0.1
